313 - Add ids home page

deep7102 opened the pull request:
**Explain the details for making this change. What existing problem does the pull request solve?**
- Added Ids Home Page component
- Added auto-fit setting in ids-cards (card should fit in as parent container width/height)

**Related github/jira issue (required)**:
Closes #313

**Steps necessary to review your pull request (required)**:
- Pull this branch and build/run the demo app
- Navigate to:
- http://localhost:4300
- http://localhost:4300/ids-home-page
- http://localhost:4300/ids-home-page/sandbox.html
- http://localhost:4300/ids-home-page/side-by-side.html
- http://localhost:4300/ids-card/auto-fit.html
- See homepage uplift styles
- Resize the browser window to different view-ports (see the columns order and sizes) 
- should pass e2e, functional tests and coverage
- See the docs and tests

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

<!-- After submitting your PR, please check back to make sure checks on the PR -->

tmcconechy commented on src/components/ids-home-page/README.MD:
````diff
@@ -0,0 +1,315 @@
+</ids-home-page>
+```
+
+No animation with cards while change/resize.
````
```suggestion
No animation with cards while changing/resizing.
```

tmcconechy commented on src/components/ids-home-page/README.MD:
```diff
@@ -0,0 +1,315 @@
+
+## Description
+
+The Home Page component will set the layout to display the cards in specific order according to available space in it's container. It take care everything as laid out in the best order and use of space it can be. To keep the responsive behavior it will readjust each card size and position on the event of resize.
```
```suggestion
Homepages hold combinations of widgets that are tailored to the needs of a user’s workflow. The Home Page component will set the layout to display the cards in specific order according to available space in it's container. It take care everything as laid out in the best order and use of space that is possible. To keep the responsive behavior it will readjust each card size and position on the event of resize.
```

tmcconechy commented on src/components/ids-home-page/README.MD:
```diff
@@ -0,0 +1,315 @@
+
+## Use Cases
+
+- Use when you want organize your content in specific order.
```
- A homepage is an admin or end-user configured page made up of widgets that are relevant to the workflow of a general role or an individual. Widgets can be resized to create layouts best-suited for the data within the widget or homepage.

deep7102 replied:
Update now -thanks

tmcconechy reviewed (commented):
Looks good , really good :) Im just wondering why the title is smaller:

![Screen Shot 2021-10-04 at 9 55 38 AM](https://user-images.githubusercontent.com/814283/135864886-7e361d8e-50da-4dcf-aae5-dc9dcc1c8e87.png)

Than on http://localhost:4300/ids-card and i added some docs.

deep7102 commented:
> Looks good , really good :) Im just wondering why the title is smaller:
> 
> ![Screen Shot 2021-10-04 at 9 55 38 AM](https://user-images.githubusercontent.com/814283/135864886-7e361d8e-50da-4dcf-aae5-dc9dcc1c8e87.png)
> 
> Than on http://localhost:4300/ids-card and i added some docs.

Thanks! updated all, and ready to review now

andriivarhanov commented on src/components/ids-home-page/ids-home-page.js:
```diff
@@ -0,0 +1,753 @@
+   * @returns {boolean} true if the value boolean
+   */
+  #isBool(val) {
+    return val === true || val === 'true' || val === false || val === 'false';
```
it seems like `stringToBool` gives same result

deep7102 replied:
@andriivarhanov cannot use [stringToBool](https://github.com/infor-design/enterprise-wc/blob/ac7fa03527dea48da50ae5c568ac2b41edf9654d/src/utils/ids-string-utils/ids-string-utils.js#L32-L37) here because it is not same output as we need
- `stringToBool` can let pass any string value as true, for example  `stringToBool(blah-blah)` will return true
- but here we only need true/false value

andriivarhanov commented on src/components/ids-home-page/ids-home-page.js:
```diff
@@ -0,0 +1,753 @@
+   */
+  set cols(value) {
+    if (value) {
+      this.setAttribute(attributes.COLS, value.toString());
```
same about `toString` everywhere where `setAttribute`

deep7102 replied:
updated now

andriivarhanov reviewed (approved):
Everything works great, there are just a couple of comments on details I've noticed. 
Also I was trying to change attributes dynamically (gap, gapX, gapY etc) and it didn't work until I noticed `refresh` method so I was thinking maybe we add some line in readme that refresh should be used after every attribute change

tmcconechy reviewed (commented):
im ready to approve once your done the final small edits. nice work on this one 👍🏻

deep7102 commented:
Thanks! @andriivarhanov updated now, all of the points you saw

deep7102 commented on src/components/ids-card/ids-card.js:
```diff
@@ -54,22 +59,39 @@ class IdsCard extends mix(IdsElement).with(IdsEventsMixin, IdsThemeMixin) {
+  set autoFit(value) {
+    const className = 'ids-card-auto-fit';
+    if (stringUtils.stringToBool(value)) {
+      this.setAttribute(attributes.AUTO_FIT, value.toString());
```
updated now

deep7102 commented on src/components/ids-card/ids-card.scss:
```diff
@@ -25,6 +25,19 @@
+
+    box-sizing: border-box;
+    min-height: auto;
+    height: 100%;
```
updated now

andriivarhanov reviewed (approved):
thanks, looks great
